subprojects/docs/src/docs/userguide/buildCache.adoc

```diff
@@ -191,90 +191,6 @@ We suggest adding tests to ensure that the task inputs are relocatable, i.e. tha
-
-| Changes in Gradle's file encoding that affect the build script
-| Gradle can produce different task output based on the file encoding used by the JVM. Gradle will use a default file encoding based on the operating system if `file.encoding` is not explicitly set.
-| Set the UTF-8 file encoding on all tasks which allow setting the encoding. Use UTF-8 file encoding everywhere by setting `file.encoding` to `UTF-8` for the Gradle JVM.
-
-| Javadoc ignores custom command-line options
-| Gradle's Javadoc task does not take into account any changes to custom command-line options.
-| You can add your custom options as input properties or disable caching of Javadoc.
-|===
-
-[[sec:task_output_caching_known_issues_caveats]]
-==== Caveats
-
-These issues may affect the number of cache hits you get between machines.
-
-.Caveats
-|===
-| Description | Impact | Workaround
-
-| Overlapping outputs between tasks
-| If two or more tasks share an output directory or files, Gradle will disable caching for these tasks when it detects an overlap.
-| Use separate output directories for each task.
-
-| Using cached C/C++ object files with absolute paths
-| When Gradle compiles C/C++ code, object files tend to have absolute paths embedded inside them. This doesn't affect their correctness, but it can interfere with debuggers that search for source code at those absolute paths.
-| Build the project from the same absolute path on every machine.
-
-| Line endings in build scripts files.
-| Gradle calculates the build cache key based on the MD5 hash of the build script contents. If the line endings are different between developers and the CI servers, Gradle will calculate different build cache keys even when all other inputs to a task are the same.
-| Check if your VCS will change source file line endings and configure it to have a consistent line ending across all platforms.
-
-| Adding new actions to cacheable tasks in a build file makes that task sensitive to unrelated changes to the build file.
-| Actions added by a plugin (from buildSrc or externally) do not have this problem because their classloader is restricted to the classpath of the plugin.
-| Avoid adding actions to cacheable tasks in a build file.
-
-| Modifying inputs or outputs during task execution.
-| It's possible to modify a task's inputs or outputs during execution in ways that change the output of a task. This breaks incremental builds and can cause problems with the build cache.
-| Use a configure task to finalize configuration for a given task. A configure task configures another task as part of its execution.
-
-| Order of input files affects outputs.
-| Some tools are sensitive to the order of its inputs and will produce slightly different output. Gradle will usually provide the order of files from the filesystem, which will be different across operating systems.
-| Provide a stable order for tools affected by order.
```

Comment:
I think we should mention that file order is important in some place. Not as a caveat here, but we should mention it. I can't think of a better place than `moreAboutTasks.adoc`, which in itself is a problem, but still.

Reply:
Most of these caveats seem like interesting information. Maybe we should add them in the build cache guide?

Reply:
Yeah, that where I looked for the caveats. And I found them there.

E.g. the ordering thing is documented [here](https://guides.gradle.org/using-build-cache/#custom_and_third_party_tasks).

Reply:
Ah, cool!